### ISSUE TITLE
Keyspace creation fails permanently if a PoisonPill is sent to the only PersistentActor

### ISSUE THREAD

**Issue by DomBlack** [akka/akka-persistence-cassandra#62]
Note: This title of this issue might be wrong, we've spent the last 3 hours try to diagnose this issue and that title is the best we can do to describe it.

Our initial start of our application sees that between the startup of the first PersistentActor and that actor becoming ready took 1 minute 22 seconds (when we allow it to complete):

```
[DEBUG] 2016-02-19T10:56:38,434 [akka.persistence.Persistence(akka://octopus)] sistence.Persistence(akka://octopus) - Create plugin: cassandra-journal akka.persistence.cassandra.journal.CassandraJournal
[DEBUG] 2016-02-19T10:56:38,463 [akka.persistence.Persistence(akka://octopus)] sistence.Persistence(akka://octopus) - Create plugin: cassandra-snapshot-store akka.persistence.cassandra.snapshot.CassandraSnapshotStore
...
[DEBUG] 2016-02-19T10:58:00,288 [akka.tcp://octopus@localhost:22452/system/cassandra-journal] e.cassandra.journal.CassandraJournal - initialized CassandraSession successfully
```

During that time the actor in question had scheduled a 15 second PoisonPill to itself to expect it's initial state. When the actor was stopped, it's guardian restarted it on a 2 second back off.

When the PoisonPill was processed it looks like the Cassandra plugin broke something (we're not sure what). Because if we stopped the system completely, increased the PoisonPill time to 10 minutes and restarted, the system never came up and eventually gave us:

```
Persistence failure when replaying events for persistenceId [TLO-B]. Last known sequence number [0]
akka.pattern.CircuitBreakerOpenException: Circuit Breaker is open; calls are failing fast
```

If we changed from the original key space name (`KS1`), to a new one (`KS2`) and started the system with the 10 minute timeout, after 1 minute 22 seconds the actor got and processed it's initial state. We never saw a persistence failure. Every time we restarted the system under `KS2` it starts up almost instantly without an issue, however restarting under `KS1` gives us the PersistenceFailure messages after a while.
#### Infrastructure

We are running a 4 node Cassandra cluster in separate VM's on a single host.
Our application sits on another VM on a different host.

**Comment by patriknw** [comment]
Sounds interesting, but hard to see what can be wrong. There are still some blocking things in the initialization of the CassandraJournal actor, and that might prevent progress if something hangs there. Just a wild guess so far. Can you take a few thread dumps when you see the problem?

**Comment by patriknw** [comment]
@DomBlack Can you try to change the dispatcher to something like:
```
  persistence-dispatcher {
    type = Dispatcher
    executor = "thread-pool-executor"
    thread-pool-executor {
      fixed-pool-size = 16
    }
    throughput = 1
  }

  cassandra-journal.plugin-dispatcher = persistence-dispatcher
  cassandra-snapshot-store.plugin-dispatcher = persistence-dispatcher
  cassandra-query-journal.plugin-dispatcher = persistence-dispatcher
```

**Comment by DomBlack** [comment]
Unfortunately I don't have the time at the moment to retest that setup, we'll make the config change, but we won't be in the position to retest the problem for a couple of weeks.

I was hoping to create a test project which demoed the issue which I could share from the point of debugging.

**Comment by patriknw** [comment]
ok, no problem

**Comment by patriknw** [comment]
@DomBlack I have released 0.12 which includes a big refactoring to eliminate the blocking. There are a few more left but those are contained with a separate dispatcher for now.
https://github.com/akka/akka-persistence-cassandra/issues?q=milestone%3A0.12

Note that the dispatcher settings have been changed. https://github.com/akka/akka-persistence-cassandra/blob/v0.12/src/main/resources/reference.conf

**Comment by chbatey** [comment]
Closing this due to age and version of the plugin it is for
